conda/activate.py

```diff
@@ -291,6 +308,28 @@ def build_activate(self, env_name_or_prefix):
 
     def build_deactivate(self):
         # query environment
+        try:
```

Comment:
Better avoid a large `try: ... except` block, i.e.,
```python
try:
    old_conda_prefix = self.environ['CONDA_PREFIX']
except KeyError:
    pass
else:
    ...
```
or (more consistent with the other code in this file)
```python
old_conda_prefix = self.environ['CONDA_PREFIX']
if old_conda_prefix is not None:
    ...
```